crypto/property/property.c

```diff
@@ -489,11 +494,13 @@ int ossl_method_store_cache_set(OSSL_METHOD_STORE *store, int nid,
         p->query = p->body;
         p->method = method;
         memcpy((char *)p->query, prop_query, len + 1);
-        if ((old = lh_QUERY_insert(alg->cache, p)) != NULL)
+        if ((old = lh_QUERY_insert(alg->cache, p)) != NULL) {
             OPENSSL_free(old);
-        if (old != NULL || !lh_QUERY_error(alg->cache)) {
-            store->nelem++;
-            if (store->nelem >= IMPL_CACHE_FLUSH_THRESHOLD)
+            ossl_property_unlock(store);
```

Comment:
So this is the actual 'fix for the infinite recursion'?

Reply:
Either of the two fix the infinite recursion.
One by removing the dependency one RAND, the other by setting the flag earlier.

Reply:
And line 499 really belongs to the RAND commit and not to the other?

Reply:
Apologies, wrong place.  Line 429 moved to 432 (which despite the number being higher is actually two lines earlier).  The `need_flush` flag was being cleared *after* the flush rather at the start of it.  Without it being cleared, the fetch of the DRBG caused another flush which fetched the DRBG again which caused another flush which ...

This spot is simplifying the code a little.

Reply:
Doesn't this belong to 9c93e33e8f8889789a249f95cf5259aba9bd03c0?